integrations/github.js

```diff
@@ -174,22 +175,45 @@ module.exports = async function(robot, kredits) {
   });
 
 
-
   if (process.env.GITHUB_KEY && process.env.GITHUB_SECRET) {
     const grantConfig = {
       defaults: {
         protocol: (process.env.GRANT_PROTOCOL || "http"),
-        host: (process.env.GRANT_HOST || 'localhost:3000')
+        host: (process.env.GRANT_HOST || 'localhost:8888'),
+        transport: 'session',
+        response: 'tokens',
+        path: '/kredits/signup'
       },
       github: {
         key: process.env.GITHUB_KEY,
         secret: process.env.GITHUB_SECRET,
         scope: ['user', 'public_repo'],
-        callback: 'https://kredits.kosmos.org/signup/github'
+        callback: '/kredits/signup/github'
       }
     };
 
-    robot.router.use('/kredits/signup/github/', grant(grantConfig));
+    const allowCORS = function (req, res, next) {
+      res.setHeader('Access-Control-Allow-Origin', '*');
+      res.setHeader('Access-Control-Allow-Methods', 'GET, POST, OPTIONS');
+      res.setHeader('Access-Control-Allow-Headers', 'Origin, Content-Type, Accept');
+      res.setHeader('Access-Control-Expose-Headers', 'ETag, Content-Length');
+      next();
+    };
+    robot.router.use(allowCORS);
+
+    robot.router.use(session({secret: 'grant'}));
+    robot.router.use('/kredits/signup', grant(grantConfig));
+
+    robot.router.get('/kredits/signup/github', async (req, res) => {
+      const access_token = req.session.grant.response.access_token;
+      const kreditsWebUrl = process.env.KREDITS_WEB_URL || 'https://kredits.kosmos.org';
+
+      res.redirect(`${kreditsWebUrl}/signup/github#access_token=${access_token}`);
```

Comment:
why don't you directly reditect to kredits-web? - using the grant callback option?

Reply:
That first redirect should go to the backend server, because that one only contains a code that you can then use together with the client secret to exchange it for the actual access token. But you shouldn't put the client secret in a client-side web app.

Reply:
Also, the endpoint to retrieve the access token using the code and client secret (POST https://github.com/login/oauth/access_token) doesn't have CORS headers, so requests from Web apps don't work at all.